### PR TITLE
V0.14.x fix matplotlib test warnings and circular import

### PR DESCRIPTION
**Pull request by hmaarrfk**
Fixes: https://github.com/scikit-image/scikit-image/issues/3631
Fixes: https://github.com/scikit-image/scikit-image/issues/3630

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](https://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/io/tests/test_mpl_imshow.py [hunk 1]
```diff
@@ -18,6 +18,12 @@ def setup():
 im_lo = imf / 1000
 im_hi = imf + 10
 
+the_expected_warnings = [r"tight_layout : falling back to Agg|\A\Z",
```
Comment (jni) [inline skimage/io/tests/test_mpl_imshow.py]: Can you rename this to something more meaningful? e.g. `expected_imshow_warnings` or `expected_warnings_on_imshow`?

Comment (jni) [reply skimage/io/tests/test_mpl_imshow.py]: Reading the later tests, I might suggests three constants:
- base expected warnings on imshow
- CObject warning on imshow
- low range warning on imshow

Then you can combine them with + in each test?

File: skimage/io/tests/test_mpl_imshow.py [hunk 2]
```diff
@@ -18,6 +18,12 @@ def setup():
...
+                         r"CObject type is marked|\A\Z",
+                         # Maptlotlib 2.2.3 seems to use np.asscalar which issues a warning
```
Comment (jni) [inline skimage/io/tests/test_mpl_imshow.py]: ... with NumPy 1.16, right?

Comment (hmaarrfk) [reply skimage/io/tests/test_mpl_imshow.py]: yea.

File: skimage/io/tests/test_mpl_imshow.py (hunk 1 above)
Comment (hmaarrfk) [reply skimage/io/tests/test_mpl_imshow.py]: this is alot of work, for a branch that isn't even master.

Comment (jni) [reply skimage/io/tests/test_mpl_imshow.py]: Fair enough, hence why I didn't tick "request changes". =) The first suggestion is not a lot of work though! Actually let me do it with suggestions.

Comment (jni) [reply skimage/io/tests/test_mpl_imshow.py]:
```suggestion
expected_imshow_warnings = [r"tight_layout : falling back to Agg|\A\Z",
```

Comment (jni) [reply skimage/io/tests/test_mpl_imshow.py]: Never mind, just saw that you took care of it! Thank you!! =)

**Comment by hmaarrfk** [comment]
So the dask bug has been cross referenced and 
I think this should be needed to at least get appveyeor a little further in the builds
https://github.com/scikit-image/scikit-image/pull/3222

**Pull request merged**